### ISSUE TITLE
go-fuzz-build fails

### ISSUE THREAD

**Issue by weissjeffm** [dvyukov/go-fuzz#19]
I am able to run the examples successfully (I tried json).

However if I try to fuzz my own project, `go-fuzz-build` fails immediately with a build error that doesn't occur with the normal go toolset (go 1.4) in my project. 

```
[jweiss@localhost gotary]$ go-fuzz-build github.com/monetas/gotary/processing/fuzz
failed to execute go build: exit status 2
# github.com/monetas/gotary/database
database/database.go:95: undefined: types.NymState
database/database.go:97: undefined: types.NymState
```

Examining my source, those two errors shouldn't happen. The `types` package is imported properly at the top of the file, `NymState` is in a file in that package, and there are other references to the `types` package earlier in the `database.go` file that do not cause complaints.

I tried keeping the `go-fuzz-build` tmp dir, and examining its contents, but there's just a trivial main.go file there and nothing else.

```go
package main

import (
	target "github.com/monetas/gotary/processing/fuzz"
	dep "github.com/dvyukov/go-fuzz/go-fuzz-dep"
)

func main() {
	dep.Main(target.Fuzz, lits)
}

var lits = []string{

}
```

I am guessing something must go wrong with copying or building somehow but I'm not sure what to do from here.

**Comment by weissjeffm** [comment]
I commented out the call to `testNormalBuild(pkg)` and rebuilt `go-fuzz-build` and then it worked.
:confused:

**Comment by dvyukov** [comment]
@weissjeffm thanks for the report, Jeff

github.com/monetas/gotary is a private repo, right? I don't see it. Or it is just not submitted?
Either way I am interested in fixing this bug.

The tmp dir contains only a single file during testNormalBuild, because your original GOPATH is still used. go-fuzz-build does somewhat weird things, but there is nothing magical. You can see it here:
https://github.com/dvyukov/go-fuzz/blob/master/go-fuzz-build/main.go#L129
It basically creates the main file, then prepends new GOPATH dir to existing GOPATH, and runs 'go build -tags gofuzz go-fuzz-main'.

Please:
1. Test on sonar branch of go-fuzz. There is at least one windows-related bug fix, which is to use os.PathListSeparator instead of ":" as GOPATH separator:
https://github.com/dvyukov/go-fuzz/blob/sonar/go-fuzz-build/main.go#L134
2. Keep the temp dir, and then run:
$ GOPATH=[TMP_DIR]:$GOPATH go build -v -tags gofuzz go-fuzz-main
Note the -v flag. What is the output? Does it pick up the file containing types.NymState? Does the file contain any '// +build' directives?

**Comment by weissjeffm** [comment]
After trying with sonar, I can no longer reproduce the issue anywhere.

I switched to sonar, ran "go install" on go-fuzz-build and go-fuzz subdirs. Then ran go-fuzz-build on my project. No errors. Then I switched back to master and did the same thing. No errors.

I guess there must have been some state stored *somewhere* that was causing the problem. I had tried "go clean" before opening this issue and that hadn't helped.
